hyper/http11/connection.py

```diff
@@ -83,17 +106,21 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Keep request methods in a queue in order to be able to know
+        # in get_response() what was the request verb.
+        self._request_methods = []
```

Comment:
Hrm, this doesn't make me much happier, as it turns out. 😞 

I think the thing to do is to keep track of *only* the currently outstanding request method. We don't allow pipelining, really, so we shouldn't have a list: just a single value. We can then use that in `get_response`. Does that sound more sensible?